libs/balances-and-allowances/src/hooks/useNativeTokenBalance.ts

```diff
@@ -9,9 +9,10 @@ const SWR_CONFIG = { refreshInterval: ms`11s` }
 
 export function useNativeTokenBalance(
   account: string | undefined,
+  chainId: number,
   swrConfig: SWRConfiguration = SWR_CONFIG,
 ): SWRResponse<BigNumber> {
-  const provider = useWalletProvider()
+  const provider = getRpcProvider(chainId)
```

Comment:
Just highlighting. Important change! 

No longer relying on the wallet for native balance! Must be sure our RPC providers are available and can handle the load.

Reply:
Thanks for paying your attention! It's important indeed.
Anyway, I've improved that part here: https://github.com/cowprotocol/cowswap/pull/5509